.github/workflows/docker-publish.yml

```diff
@@ -16,17 +16,19 @@
 name: docker-publish
 
 on:
-  push:
-    branches: [ "master" ]
-    tags: [ 'v*.*.*' ]
+  release:
```

Comment:
why is release？

Reply:
I found images are only pushed to dockerhub when release. So I'm wondering if I need to do the same when it comes to ghcr.io. Then I replace "push branches master" with "release" here.

If "push branches master" is intensinally-designed, I'll fix it.

Reply:
When releasing a version, we push the docker image, you can refer to it here: https://shenyu.apache.org/community/release-guide#finish-publishing

Not used github action.

Reply:
@yu199195 We can use actions to help us build and publish images that support multiple platforms, such as: arm64, amd64, which can make it easier for us to publish, and currently some other apache projects do the same.

And apache infra seems to provide the docker hub account and password in the GitHub settings: https://cwiki.apache.org/confluence/display/INFRA/Github+Secrets+and+Tokens

Reply:
@yu199195 @JooKS-me I restore the former docker-publish workflow, and create a new one to publish images to dockerhub in recent new commit.